float64 distance
---
bool ret
